packages/astro/src/jsx/babel.ts

```diff
@@ -299,11 +297,8 @@ export default function astroJSX(): PluginObj {
 					}
 					let resolvedPath: string;
 					if (meta.path.startsWith('.')) {
-						const fileURL = pathToFileURL(state.filename!);
-						resolvedPath = `/@fs${new URL(meta.path, fileURL).pathname}`;
-						if (resolvedPath.endsWith('.jsx')) {
-							resolvedPath = resolvedPath.slice(0, -4);
-						}
+						resolvedPath = resolve(meta.path, state.filename!);
+						resolvedPath = resolveClientDevPath(resolvedPath);
 					} else {
```

Comment:
This code runs in node, but uses the URL api which auto encodes paths by default. Switch to using `path.resolve` and handle the `.jsx` condition with `resolveClientDevPath`.

I also removed the `/@fs` part as tracing through it's usage, this isn't needed. This path is only used to pass as Vite entry, which can read from the filesystem directly as absolute path.

Reply:
I reverted this change. I thought `resolvedPath` meant the filesystem path, but it seems to be treated as a URL that starts with `/@fs` everywhere.